pkg/controller/plan/validation.go

```diff
@@ -667,34 +677,183 @@ func (r *Reconciler) validateVddkImage(plan *api.Plan) (err error) {
 		Category: Critical,
 		Message:  "VDDK image is necessary for this type of migration",
 	}
+	vddkInvalid := libcnd.Condition{
+		Type:     VDDKInvalid,
+		Status:   True,
+		Reason:   NotSet,
+		Category: Critical,
+		Message:  "VDDK init image is invalid",
+	}
 
 	source := plan.Referenced.Provider.Source
 	if source == nil {
-		return nil
+		return
 	}
 	destination := plan.Referenced.Provider.Destination
 	if destination == nil {
-		return nil
+		return
 	}
 
 	if source.Type() != api.VSphere {
-		// VDDK is used for other provider types
-		return nil
+		// VDDK is not used for other provider types
+		return
+	}
+
+	el9, err := plan.VSphereUsesEl9VirtV2v()
+	if err != nil {
+		return
+	}
+
+	image, found := source.Spec.Settings[api.VDDK]
+	switch {
+	case !el9 && !found:
+		// VDDK image is required when EL8 virt-v2v image is in use
+		plan.Status.SetCondition(vddkNotConfigured)
+	case found:
+		var job *batchv1.Job
+		if job, err = r.ensureVddkImageValidationJob(plan, el9, image); err != nil {
+			break
+		}
+		if len(job.Status.Conditions) == 0 {
+			err = liberr.New("validation of VDDK job is in progress")
+		}
+		for _, condition := range job.Status.Conditions {
+			switch condition.Type {
+			case batchv1.JobComplete:
+				r.Log.Info("validate VDDK job completed", "image", image)
+				err = nil
+				return
+			case batchv1.JobFailed:
+				plan.Status.SetCondition(vddkInvalid)
+				err = nil
+				return
+			default:
+				err = liberr.New("validation of VDDK job has an unexpected condition", "type", condition.Type)
+			}
+		}
+	}
+
+	return
+}
+
+func (r *Reconciler) ensureVddkImageValidationJob(plan *api.Plan, el9 bool, vddkImage string) (*batchv1.Job, error) {
+	jobLabels := getVddkImageValidationJobLabels(plan)
+	jobs := &batchv1.JobList{}
+	ctx, _ := plancontext.New(r, plan, r.Log)
+	err := ctx.Destination.Client.List(
+		context.TODO(),
+		jobs,
+		&client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(jobLabels),
+			Namespace:     plan.Spec.TargetNamespace,
+		},
+	)
+	switch {
+	case err != nil:
+		return nil, err
+	case len(jobs.Items) == 0:
+		job := createVddkCheckJob(plan, jobLabels, el9, vddkImage)
+		err = ctx.Destination.Client.Create(context.Background(), job, &client.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+		return job, nil
+	default:
+		return &jobs.Items[0], nil
 	}
+}
 
-	el9, el9Err := plan.VSphereUsesEl9VirtV2v()
-	if el9Err != nil {
-		return el9Err
+func getVddkImageValidationJobLabels(plan *api.Plan) map[string]string {
+	image := plan.Referenced.Provider.Source.Spec.Settings[api.VDDK]
+	sum := md5.Sum([]byte(image))
+	return map[string]string{
+		"plan": string(plan.ObjectMeta.UID),
+		"vddk": hex.EncodeToString(sum[:]),
```

Comment:
what's the benefit of the md5? wouldn't the name as is and the plan will be enough?

Reply:
the name would be enough but it typically contains characters that are invalid in this context

Reply:
it will be also nicer to "debug" it without encoding on some sort. because then, we can look into the job and based on the label see the real path of the image. however if it requires too much change in terms of invalid chars... that troublesome

Reply:
the vddk image is set in the annotations for debugging (see line 815 below) - the labels are purely for matching